test/unit/mdc-list/foundation.test.js

```diff
@@ -383,7 +392,197 @@ test('#handleKeydown does not find ancestor with mdc-list-item so returns early'
+  td.when(mockAdapter.getListItemCount()).thenReturn(3);
+  td.when(mockAdapter.isListItem(target)).thenReturn(true);
+  foundation.setSingleSelection(true);
+  foundation.handleKeydown(event);
+
+  td.verify(preventDefault(), {times: 1});
+  td.verify(mockAdapter.setAttributeForElementIndex(0, strings.ARIA_SELECTED, true));
+});
+
+test('#handleKeydown space key is triggered 2x when singleSelection is true un-selects the list item', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const preventDefault = td.func('preventDefault');
+  const target = {classList: ['mdc-list-item']};
+  const event = {key: 'Space', target, preventDefault};
+
+  td.when(mockAdapter.getFocusedElementIndex()).thenReturn(0);
+  td.when(mockAdapter.getListItemCount()).thenReturn(3);
+  td.when(mockAdapter.isListItem(target)).thenReturn(true);
+  foundation.setSingleSelection(true);
+  foundation.handleKeydown(event);
+  foundation.handleKeydown(event);
+
+  td.verify(preventDefault(), {times: 2});
+  td.verify(mockAdapter.removeAttributeForElementIndex(0, strings.ARIA_SELECTED));
+});
+
+test('#handleKeydown space key is triggered when singleSelection is true on second ' +
+  'element updates first element tabindex', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const preventDefault = td.func('preventDefault');
+  const target = {classList: ['mdc-list-item']};
+  const event = {key: 'Space', target, preventDefault};
+
+  td.when(mockAdapter.getFocusedElementIndex()).thenReturn(1);
+  td.when(mockAdapter.getListItemCount()).thenReturn(3);
+  td.when(mockAdapter.isListItem(target)).thenReturn(true);
+  foundation.setSingleSelection(true);
+  foundation.handleKeydown(event);
+
+  td.verify(preventDefault(), {times: 1});
+  td.verify(mockAdapter.setAttributeForElementIndex(1, 'tabindex', 0));
+});
+
+test('#handleKeydown space key is triggered 2x when singleSelection is true on second ' +
+  'element updates first element tabindex', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const preventDefault = td.func('preventDefault');
+  const target = {classList: ['mdc-list-item']};
+  const event = {key: 'Space', target, preventDefault};
+
+  td.when(mockAdapter.getFocusedElementIndex()).thenReturn(1);
+  td.when(mockAdapter.getListItemCount()).thenReturn(3);
+  td.when(mockAdapter.isListItem(target)).thenReturn(true);
+  foundation.setSingleSelection(true);
+  foundation.handleKeydown(event);
+  foundation.handleKeydown(event);
+
+  td.verify(preventDefault(), {times: 2});
+  td.verify(mockAdapter.setAttributeForElementIndex(0, 'tabindex', 0));
+  td.verify(mockAdapter.setAttributeForElementIndex(1, 'tabindex', -1));
+});
+
+test('#handleKeydown space key is triggered and focused is moved to a different element', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const preventDefault = td.func('preventDefault');
+  const target = {classList: ['mdc-list-item']};
+  const event = {key: 'Space', target, preventDefault};
```

Comment:
You don't have any tests events with keyCodes for space and enter. Will that increase coverage?

Reply:
It doesn't doesn't increase coverage since that statement is already covered by the `key` parameter.